src/core/ext/xds/xds_http_rbac_filter.cc

```diff
@@ -546,12 +546,15 @@ ChannelArgs XdsHttpRbacFilter::ModifyChannelArgs(
 absl::StatusOr<XdsHttpFilterImpl::ServiceConfigJsonEntry>
 XdsHttpRbacFilter::GenerateServiceConfig(
     const FilterConfig& hcm_filter_config,
-    const FilterConfig* filter_config_override) const {
+    const FilterConfig* filter_config_override,
+    absl::string_view filter_name) const {
   Json policy_json = filter_config_override != nullptr
                          ? filter_config_override->config
                          : hcm_filter_config.config;
-  // The policy JSON may be empty, that's allowed.
-  return ServiceConfigJsonEntry{"rbacPolicy", JsonDump(policy_json)};
+  auto json_object = policy_json.object();
+  json_object.emplace("filter_name", std::string(filter_name));
```

Comment:
style question - should `filter_name` and `rbacPolicy` be consts?

Reply:
The existing pattern doesn't require us to do this here. I also don't think it's strictly necessary since there isn't a whole lot repetitions.